### DISCUSSION TITLE
Sveltekit + Pocketbase + Marmot = ❤️

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#3833] [Show and tell]
Hey everyone,

I've successfully scaled my SvelteKit app horizontally using Pocketbase and Marmot, and it's been smooth sailing.

My micro-SaaS application is a stock analysis platform where users can sign up, create their watchlist, and leverage AI signals for bullish or bearish stocks, all supported by thorough backtesting results.

Additionally, I've developed a community hub similar to Reddit/9GAG where users can share memes, comment, and like posts. This entire venture has been a solo project, and I must say, this setup has worked wonders for scaling my Svelte project.

If you're curious or have any questions, feel free to reach out! 😊

![Screenshot from 2023-11-28 00-11-21](https://github.com/pocketbase/pocketbase/assets/22890183/4e82a1fe-463e-4db0-8d2f-d4b91ea28709)

Website: https://stocknear.com/

**Comment by sbusso** [comment]
Nice product @MuslemRahimi . Could you share more of your experience with scaling with Marmo, what kind of traffic you can handle, and what was the limit of the scaling version?

**Reply by MuslemRahimi** [reply]
We are still very early in the development so I am still not sure how stable everything is.

**Comment by tigawanna** [comment]
Are the pocketbase instances aware of each other or is it all behind a reverse proxy?

**Reply by MuslemRahimi** [reply]
They are connected with each other without a proxy server

**Reply by laszlo1337** [reply]
Curious why did you need multiple pocketbase instances? I am developing an app using PB myself and my plan was to use PM2 to spawn 1 PB instance and 3 SvelteKit app instances. My Reasoning was that I assumed PB is multi-threaded, and NodeJS based apps are not. So on 3 core machine that would be my setup. Am I wrong about PB being multi-threaded? Couldn't really find any resources about that

**Reply by tigawanna** [reply]
Using marmot the pocketbase instances can be in separate regions / machines , so that in case of one failing the others pick up and when the failed one comes back online it's db gets caught up to the rest

**Reply by tigawanna** [reply]
> They are connected with each other without a proxy server

Do you have like a config of how you did it ? Last time I checked on marmot you had to specify each instances IP , meaning you needed a reverse proxy that the client would reach out to and have the request get sent to these IPs

**Reply by MuslemRahimi** [reply]
Sorry then i misunderstood you. Yes they are reverse proxy connected
